src/node/txreconciliation.cpp

```diff
@@ -125,6 +133,43 @@ class TxReconciliationTracker::Impl
         return ReconciliationRegisterResult::SUCCESS;
     }
 
+    bool AddToSet(NodeId peer_id, const uint256& wtxid) EXCLUSIVE_LOCKS_REQUIRED(!m_txreconciliation_mutex)
+    {
+        AssertLockNotHeld(m_txreconciliation_mutex);
+        LOCK(m_txreconciliation_mutex);
+        if (!IsPeerRegistered(peer_id)) return false;
+        auto& recon_state = std::get<TxReconciliationState>(m_states.find(peer_id)->second);
+
+        // Check if reconciliation set is not at capacity for two reasons:
+        // - limit sizes of reconciliation sets and short id mappings;
+        // - limit CPU use for sketch computations.
+        //
+        // Since we reconcile frequently, reaching capacity either means:
+        // (1) a peer for some reason does not request reconciliations from us for a long while, or
+        // (2) really a lot of valid fee-paying transactions were dumped on us at once.
+        // We don't care about a laggy peer (1) because we probably can't help them even if we fanout transactions.
+        // However, exploiting (2) should not prevent us from relaying certain transactions.
+        //
+        // Transactions which don't make it to the set due to the limit are announced via fan-out.
+        if (recon_state.m_local_set.size() >= MAX_SET_SIZE) return false;
```

Comment:
I think can be `>` only. Generally sounds maximum in net_processing is understood as strictly superior, e.g `MAX_INV_SZ` usage.

Reply:
This happens before the addition. Your suggestion would mean that the size could be 3001, which is not desired.